### DISCUSSION TITLE
model is required

### DISCUSSION THREAD

**Discussion by Aleksi-learn-code** [awaescher/OllamaSharp#90] [General]
Hello, I don't understand, I have a code for practice:
`using System;
using System.Windows.Forms;
using System.Threading;
using System.Threading.Tasks;
using OllamaSharp;
using OllamaSharp.Models.Chat;

namespace OllamaChatApp
{
    public partial class Form1 : Form
    {
        private Chat _chat; // Entferne readonly

        public Form1()
        {
            InitializeComponent();
            InitializeOllama();
        }

        private void InitializeOllama()
        {
            var ollamaApiClient = new OllamaApiClient("http://localhost:11434");
            _chat = new Chat(ollamaApiClient, "llama2");
        }

        private async void buttonsend_Click(object sender, EventArgs e)
        {
            if (string.IsNullOrWhiteSpace(richTextBoxinput.Text))
                return;

            string userInput = richTextBoxinput.Text;
            richTextBoxoutput.AppendText("You: " + userInput + Environment.NewLine);
            richTextBoxinput.Clear();

            richTextBoxoutput.AppendText("Assistant: ");
            buttonsend.Enabled = false;

            try
            {
                // Verwende die Methode Send für den Chat-Prozess
                var response = await _chat.Send(userInput, CancellationToken.None).StreamToEnd();
                richTextBoxoutput.AppendText(response);  // Zeigt die vollständige Antwort an
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
                MessageBox.Show($"Error: {ex.Message}", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
            finally
            {
                richTextBoxoutput.AppendText(Environment.NewLine + Environment.NewLine);
                buttonsend.Enabled = true;
            }
        }
    }
}
`
OllamaSharp 3.0.8  Nuget
What am I doing wrong? Can you please help me?

**Comment by Aleksi-learn-code** [comment]
var ollamaApiClient = new OllamaApiClient("http://localhost:11434");
            ollamaApiClient.SelectedModel = "llama3"; // Set Modell
           ollamaApiClient.
            _chat = new Chat(ollamaApiClient);

**Comment by awaescher** [comment]
I guess you missed to tell us the error that occurs. Maybe Ollama doesn't have the model downloaded locally.

**Reply by Aleksi-learn-code** [reply]
Sorry, the error was “model required”. I was able to solve this using the above code in the initialization
